controllers/cassandracluster/statefulset.go

```diff
@@ -168,13 +170,32 @@ func statefulSetsAreEqual(sts1, sts2 *appsv1.StatefulSet) bool {
 	sts2.Status.Replicas = sts1.Status.Replicas
 
 	if patchResult, err := patch.DefaultPatchMaker.Calculate(sts1, sts2); err != nil || !patchResult.IsEmpty() {
-		logrus.Debug("Template is different: " + pretty.Compare(sts1.Spec, sts2.Spec))
+		if logrus.IsLevelEnabled(logrus.DebugLevel) {
+			logTemplateChange(sts1, sts2, patchResult.Patch)
+		}
 		return false
 	}
 
 	return true
 }
 
+func logTemplateChange(sts1 *appsv1.StatefulSet, sts2 *appsv1.StatefulSet, patchResult []byte) {
+	logrus.Debug("Template is different: " + pretty.Compare(sts1.Spec, sts2.Spec))
+	logrus.Debug("Generated patch is: " + tryJsonPrettyPrint(patchResult))
+}
+
+func tryJsonPrettyPrint(patchResult []byte) string {
```

Comment:
I think we can use cmp.Diff() for this from "github.com/google/go-cmp/cmp" - it gives nice diff output.

Reply:
I've just checked it and indeed, output looks nice.

However, it has some cons:
- we would print diff generated with different method than diff used to decide whether update statefulset or not - this may lead to confusion
- cmp.Diff knows nothing about "banzaicloud.com/last-applied" annotation (which is used to generate three way merge patch) - this may lead to printing incorrect diff - e.g. adding all unnecessary k8s defaults to diff output such as RevisionHistoryLimit=10 etc.